### ISSUE TITLE
Right-click -> FindReferences when used in an interface, returns too many useless results

### ISSUE THREAD

**Issue by knocte** [mono/monodevelop#4709]
Steps to reproduce (testcase which definitely leads to the bug, not sure if it can be simpler):
1. Clone https://github.com/metacosa/nbitcoin
2. Open it in VisualStudioForMac.
3. Go to Transaction.fs file, to the TxOut constructor that takes arguments `(Money value, IDestination destination)`.
4. Right-click on the `IDestination` word, -> Find References.

Expected results:
It should show just results where this interface is used, e.g. as a type declaration, or the classes that implement the interface, etc.

Current results:
It shows too many results, e.g. when all types that inherit from IDestination are used, such as `txbuilder.Send(new Key().ScriptPubKey, Money.Coins(0.4m));` (Key implements IDestination).

Version information:
```
Visual Studio Community 2017 for Mac
Version 7.4.3 (build 10)
Installation UUID: fd06ff65-69de-47ae-8b99-ddf257032b7c
Runtime:
	Mono 5.8.1.0 (2017-10/6bf3922f3fd) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)

	Package version: 508010000

NuGet
Version: 4.3.1.4445

.NET Core
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.5
	2.0.0
SDK: /usr/local/share/dotnet/sdk/2.1.4/Sdks
SDK Versions:
	2.1.4
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.8.1/lib/mono/msbuild/15.0/bin/Sdks

Xamarin.Profiler
Version: 1.6.1
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Apple Developer Tools
Xcode 9.3 (14154)
Build 9E145

Xamarin.Mac
Version: 4.2.1.29 (Visual Studio Community)

Xamarin.iOS
Version: 11.9.1.24 (Visual Studio Community)
Hash: f62de472
Branch: xcode9.3
Build date: 2018-03-29 19:30:53-0400

Xamarin.Android
Version: 8.2.0.16 (Visual Studio Community)
Android SDK: /Users/andres/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.4 (API level 19)
		7.1 (API level 25)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 26.0.2
SDK Build Tools Version: 26.0.3

Java SDK: /usr
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Xamarin Inspector
Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

Build Information
Release ID: 704030010
Git revision: 5af3e98549653fcc5335896ccc296343d08f31bb
Build date: 2018-04-12 12:32:07-04
Xamarin addins: fadb82fb0fe2668cb7789f8b524aec6fb8568ee7
Build lane: monodevelop-lion-d15-6

Operating System
Mac OS X 10.13.4
Darwin 17.5.0 Darwin Kernel Version 17.5.0
    Mon Mar  5 22:24:32 PST 2018
    root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64

Enabled user installed addins
Internet of Things (IoT) development (Preview) 7.1
```

> VS bug [#610540](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/610540)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.6.0.775 (master)

Pull Request #4732 merged by: Lluis Sanchez
Author: mono
Commit: bf5ed3072ca09a3af6bbab95de86a34e35a9ed2b (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #4732 merged by: Lluis Sanchez
Author: mono
Commit: bf5ed3072ca09a3af6bbab95de86a34e35a9ed2b (xamarin/monodevelop)
